### PR TITLE
api: verify Location constraint for make bucket.

### PR DESCRIPTION
**Pull request by hackintoshrao**

### CODE REVIEW

File: object-utils.go [hunk 1]
```diff
@@ -93,3 +95,36 @@ func IsValidObjectPrefix(object string) bool {
 func pathJoin(path1 string, path2 string) string {
 	return strings.TrimSuffix(path1, slashPathSeparator) + slashPathSeparator + path2
 }
+
+// validates location constraint from the request body.
+// the location value in the request body should match the Region in serverConfig.
+// other values of location are not accepted.
+// make bucket fails in such cases.
+func isValidLocationContraint(r *http.Request, serverConfigRegion string) APIErrorCode {
```
Comment (harshavardhana) [inline object-utils.go]: just take r.Body as input not httpRequest.

File: object-utils.go [hunk 2]
```diff
@@ -93,3 +95,36 @@ func IsValidObjectPrefix(object string) bool {
...
+			// Failed due to malformed configuration.
+			errCode = ErrMalformedXML
+			//writeErrorResponse(w, r, ErrMalformedXML, r.URL.Path)
+		}
+	} else {
+		// Region obtained from the body.
+		// It should be equal to Region in serverConfig.
+		// Else ErrInvalidRegion returned.
+		// For empty value location will be to set to  default value from the serverConfig.
+		if serverConfigRegion != locationContraint.Location && locationContraint.Location != "" {
```
Comment (harshavardhana) [inline object-utils.go]: Move the check 

```
if locationConstraint.Location != "" && locationConstraint.Location != serverConfigRegion {
      errCode= ErrInvalidRegion
}
```

File: object-utils.go (hunk 1 above)
Comment (harshavardhana) [inline object-utils.go]: Change the variable name of "serverConfigRegion" to just "serverRegion"

File: object-utils.go (hunk 2 above)
Comment (hackintoshrao) [inline object-utils.go]: you mean to do the check in the caller, rather than in this function ?

Comment (hackintoshrao) [inline object-utils.go]: Done.

File: object-utils.go (hunk 1 above)
Comment (hackintoshrao) [inline object-utils.go]: Done.

**Pull request merged**